src/hotspot/share/gc/shared/gcLocker.inline.hpp

```diff
@@ -29,29 +29,35 @@
 
 #include "runtime/javaThread.inline.hpp"
 
-void GCLocker::lock_critical(JavaThread* thread) {
+void GCLocker::enter(JavaThread* thread) {
+  assert(thread == JavaThread::current(), "Must be this thread");
+
   if (!thread->in_critical()) {
-    if (needs_gc()) {
-      // jni_lock call calls enter_critical under the lock so that the
-      // global lock count and per thread count are in agreement.
-      jni_lock(thread);
-      return;
+    thread->enter_critical();
+    OrderAccess::storeload();
+    if (Atomic::load(&_is_gc_request_pending)) {
+      thread->exit_critical();
+      // slow-path
```

Comment:
```suggestion
```

Not sure what this `slow-path` comment helps with. Maybe it is describing the next method (but it is named very similarly), or this is an attempt to describe the true-block of the if.
In the latter case, it would maybe be better to put this comment at the start of the true-block of the if, and say something more descriptive like `// Another thread is requesting gc, enter slow path.`

Not sure, feel free to ignore, it's just that to me the comment should either be removed or put upwards a line.